library/oneview_server_profile.py

```diff
@@ -289,6 +291,37 @@ def __present(self, data, resource):
 
         return created, changed, msg, resource
 
+    # Removes .mac entries from resource os_custom_attributes if no .mac passed into data params.
+    # Swaps True values for 'true' string, and False values for 'false' string to avoid common user errors.
+    def __validations_for_os_custom_attributes(self, merged_data, resource):
+        attributes_merged = merged_data.get('osDeploymentSettings', {}).get('osCustomAttributes', None)
+        attributes_resource = resource.get('osDeploymentSettings', {}).get('osCustomAttributes', None)
+        mac_positions_in_merged_data = self.__find_in_array_of_hashes(attributes_merged, '.mac', -4)
+        mac_positions_in_resource = self.__find_in_array_of_hashes(attributes_resource, '.mac', -4)
```

Comment:
You need to chef if it is a NIC type custom attribute in the resource, because the user **can** create a custom resource of type String with `*.mac` and it not necessarily will be filled automatically by the build plan